### PR TITLE
qbft to not include round number in proposal block

### LINKED ISSUE #2134: Qbft proposal to not include round number
Change the Qbft proposal block to not include round number and instead only populate the round number in the extra data on the commit phase before the commit seal is calculated. This is to match the approach taken in GoQuorum for the round number in the block.

### PR DESCRIPTION
**Pull request by jframe**
Signed-off-by: Jason Frame <jasonwframe@gmail.com>

## PR description
Change QBFT so that the proposal block no longer includes the round number and uses the default value of 0 in the extra data. Round number is populated before the commit seal is calculated and included in the final block.

## Fixed Issue(s)

fixes #2134 

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by usmansaleem — APPROVED** [review]
LGTM

File: consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/ProposalPayloadValidatorTest.java [hunk 1]
```diff
@@ -167,30 +167,6 @@ public void validationFailsWhenMessageMismatchesExpectedHeight() {
     verifyNoMoreInteractions(blockValidator);
   }
 
-  @Test
```
Comment (saltiniroberto) [inline consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/ProposalPayloadValidatorTest.java:170]: What about having a test that shows that we now validate Proposal messages independently of the round number in the block?

File: consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/RoundChangeMessageValidatorTest.java [hunk 2]
```diff
@@ -227,35 +227,6 @@ public void prepareFromNonValidatorFails() {
     assertThat(messageValidator.validate(message)).isFalse();
   }
 
-  @Test
```
Comment (saltiniroberto) [inline consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/RoundChangeMessageValidatorTest.java:230]: What about testing that the round number in the block does not affect the validation?

**Review by saltiniroberto — APPROVED** [review]
LGTM.

File: consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/ProposalPayloadValidatorTest.java (hunk 1 above)
Comment (jframe) [reply consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/ProposalPayloadValidatorTest.java:170]: done

File: consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/RoundChangeMessageValidatorTest.java (hunk 2 above)
Comment (jframe) [reply consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validation/RoundChangeMessageValidatorTest.java:230]: done

**Pull request merged**